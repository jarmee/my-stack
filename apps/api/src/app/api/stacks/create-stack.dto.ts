import { ApiProperty } from "@nestjs/swagger";

export class CreateStackDto {
  @ApiProperty({
    description: "Title",
    type: String,
  })
  readonly title: string;
}
