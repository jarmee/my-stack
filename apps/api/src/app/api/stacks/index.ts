export * from './stacks.module';
export * from './stack.entity';
