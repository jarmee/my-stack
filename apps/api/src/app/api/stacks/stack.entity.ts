import { Column, Entity, OneToMany, PrimaryGeneratedColumn } from 'typeorm';

import type { StackTechnology } from './technologies';

@Entity({ name: 'stacks' })
export class Stack {
  @PrimaryGeneratedColumn()
  id: number;

  @Column({ length: 256 })
  title: string;

  @OneToMany('StackTechnology', 'stack', {
    eager: true,
  })
  technologies: StackTechnology[];
}
