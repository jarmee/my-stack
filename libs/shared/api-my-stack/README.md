# shared-api-my-stack

This library was generated with [Nx](https://nx.dev).

## Running unit tests

Run `nx test shared-api-my-stack` to execute the unit tests.
