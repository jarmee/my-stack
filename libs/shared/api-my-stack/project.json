{
  "name": "shared-api-my-stack",
  "$schema": "../../../node_modules/nx/schemas/project-schema.json",
  "projectType": "library",
  "sourceRoot": "libs/shared/api-my-stack/src",
  "prefix": "my-stack",
  "targets": {
    "test": {
      "executor": "@nrwl/jest:jest",
      "outputs": [
        "{workspaceRoot}/coverage/{projectRoot}"
      ],
      "options": {
        "jestConfig": "libs/shared/api-my-stack/jest.config.ts",
        "passWithNoTests": true
      }
    },
    "lint": {
      "executor": "@nrwl/linter:eslint",
      "outputs": [
        "{options.outputFile}"
      ],
      "options": {
        "lintFilePatterns": [
          "libs/shared/api-my-stack/**/*.ts",
          "libs/shared/api-my-stack/**/*.html"
        ]
      }
    }
  },
  "tags": [
    "scope:shared",
    "type:api"
  ]
}
