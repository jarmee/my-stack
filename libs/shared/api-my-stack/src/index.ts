export * from "./lib/shared-api-my-stack.module";
