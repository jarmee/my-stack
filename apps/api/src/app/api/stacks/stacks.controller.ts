import {
  Body,
  Controller,
  Delete,
  Get,
  HttpCode,
  Param,
  Post,
  Put,
} from '@nestjs/common';
import { ApiOperation, ApiResponse, ApiTags } from '@nestjs/swagger';

import { API } from '../../constants';
import { CreateStackDto } from './create-stack.dto';
import { Stack } from './stack.entity';
import { StacksService } from './stacks.service';
import { UpdateStackDto } from './update-stack.dto';

@Controller(API.ENDPOINTS.STACKS)
@ApiTags(API.ENDPOINTS.STACKS)
export class StacksController {
  constructor(private _service: StacksService) {}

  @Post()
  @ApiOperation({ operationId: 'createStack', description: 'creates a stack' })
  @ApiResponse({
    status: 200,
    type: Stack,
  })
  create(@Body() createStack: CreateStackDto): Promise<Stack> {
    return this._service.create(createStack);
  }

  @Get()
  @ApiOperation({
    operationId: 'getAllStacks',
    description: 'returns all stacks',
  })
  @ApiResponse({
    status: 200,
    isArray: true,
    type: Stack,
  })
  getAll(): Promise<Array<Stack>> {
    return this._service.getAll();
  }

  @Get(':id')
  @ApiOperation({
    operationId: 'getStackById',
  })
  @ApiResponse({
    status: 200,
    type: Stack,
  })
  getById(@Param('id') id: number): Promise<Stack> {
    return this._service.getById(id);
  }

  @Put(':id')
  @ApiOperation({
    operationId: 'updateStack',
  })
  @ApiResponse({
    status: 200,
    type: Stack,
  })
  update(
    @Param('id') id: number,
    @Body() updateStack: UpdateStackDto
  ): Promise<Stack> {
    return this._service.update(id, updateStack);
  }

  @Delete(':id')
  @ApiOperation({
    operationId: 'deleteStack',
  })
  @ApiResponse({
    status: 204,
  })
  @HttpCode(204)
  delete(@Param('id') id: number): Promise<void> {
    return this._service.delete(id);
  }
}
