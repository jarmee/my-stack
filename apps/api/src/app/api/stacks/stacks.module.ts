import { Module } from "@nestjs/common";
import { TypeOrmModule } from "@nestjs/typeorm";

import { Stack } from "./stack.entity";
import { StacksController } from "./stacks.controller";
import { StacksService } from "./stacks.service";
import { StacksTechnologiesModule } from "./technologies";

@Module({
  imports: [TypeOrmModule.forFeature([Stack]), StacksTechnologiesModule],
  controllers: [StacksController],
  providers: [StacksService],
})
export class StacksModule {}
