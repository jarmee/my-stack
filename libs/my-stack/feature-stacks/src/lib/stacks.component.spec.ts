import { provideHttpClient } from '@angular/common/http';
import { importProvidersFrom } from '@angular/core';
import { ComponentFixture, TestBed } from '@angular/core/testing';
import { ApiModule, StacksService } from '@my-stack/shared/api-my-stack';
import { of } from 'rxjs';

import { StacksComponent } from './stacks.component';

describe('StacksComponent', () => {
  let component: StacksComponent;
  let fixture: ComponentFixture<StacksComponent>;
  let service: StacksService;

  beforeEach(async () => {
    await TestBed.configureTestingModule({
      imports: [StacksComponent],
      providers: [provideHttpClient(), importProvidersFrom(ApiModule)],
    }).compileComponents();
  });

  beforeEach(() => {
    service = TestBed.inject(StacksService);
    service.getAllStacks = jest.fn().mockReturnValue(
      of([
        {
          id: 0,
          title: 'test-0',
          technologies: [],
        },
        {
          id: 1,
          title: 'test-1',
          technologies: [],
        },
        {
          id: 2,
          title: 'test-2',
          technologies: [],
        },
        {
          id: 3,
          title: 'test-3',
          technologies: [],
        },
        {
          id: 4,
          title: 'test-4',
          technologies: [],
        },
        {
          id: 5,
          title: 'test-5',
          technologies: [],
        },
        {
          id: 6,
          title: 'test-6',
          technologies: [],
        },
        {
          id: 7,
          title: 'test-7',
          technologies: [],
        },
        {
          id: 8,
          title: 'test-8',
          technologies: [],
        },
        {
          id: 9,
          title: 'test-9',
          technologies: [],
        },
      ])
    );
  });

  beforeEach(() => {
    fixture = TestBed.createComponent(StacksComponent);
    component = fixture.componentInstance;
    fixture.detectChanges();
  });

  it('should create', () => {
    expect(component).toBeTruthy();
  });

  it('should match the snapshot', () => {
    expect(fixture).toMatchSnapshot();
  });
});
