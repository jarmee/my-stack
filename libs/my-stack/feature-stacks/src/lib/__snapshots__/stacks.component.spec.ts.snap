// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StacksComponent should match the snapshot 1`] = `
<my-stack-stacks
  __ngContext__={[Function Number]}
>
  <my-stack-stacks-overview>
    <div
      class="px-4"
    >
      <h1>
        Stacks
      </h1>
      <mat-divider
        aria-orientation="horizontal"
        class="mat-divider mat-divider-horizontal"
        role="separator"
      />
      <div
        class="grid xl:grid-cols-8 lg:grid-cols-6 md:grid-cols-4 sm:grid-cols-3 grid-cols-2 gap-4 mt-3 h-100"
      >
        <my-stack-stack>
          <div
            class="flex flex-col items-center justify-items-center cursor-pointer"
          >
            <mat-card
              class="mat-mdc-card mdc-card h-[144px] w-[144px]"
            />
            <span
              class="mt-2"
            >
              test-0
            </span>
          </div>
        </my-stack-stack>
        <my-stack-stack>
          <div
            class="flex flex-col items-center justify-items-center cursor-pointer"
          >
            <mat-card
              class="mat-mdc-card mdc-card h-[144px] w-[144px]"
            />
            <span
              class="mt-2"
            >
              test-1
            </span>
          </div>
        </my-stack-stack>
        <my-stack-stack>
          <div
            class="flex flex-col items-center justify-items-center cursor-pointer"
          >
            <mat-card
              class="mat-mdc-card mdc-card h-[144px] w-[144px]"
            />
            <span
              class="mt-2"
            >
              test-2
            </span>
          </div>
        </my-stack-stack>
        <my-stack-stack>
          <div
            class="flex flex-col items-center justify-items-center cursor-pointer"
          >
            <mat-card
              class="mat-mdc-card mdc-card h-[144px] w-[144px]"
            />
            <span
              class="mt-2"
            >
              test-3
            </span>
          </div>
        </my-stack-stack>
        <my-stack-stack>
          <div
            class="flex flex-col items-center justify-items-center cursor-pointer"
          >
            <mat-card
              class="mat-mdc-card mdc-card h-[144px] w-[144px]"
            />
            <span
              class="mt-2"
            >
              test-4
            </span>
          </div>
        </my-stack-stack>
        <my-stack-stack>
          <div
            class="flex flex-col items-center justify-items-center cursor-pointer"
          >
            <mat-card
              class="mat-mdc-card mdc-card h-[144px] w-[144px]"
            />
            <span
              class="mt-2"
            >
              test-5
            </span>
          </div>
        </my-stack-stack>
        <my-stack-stack>
          <div
            class="flex flex-col items-center justify-items-center cursor-pointer"
          >
            <mat-card
              class="mat-mdc-card mdc-card h-[144px] w-[144px]"
            />
            <span
              class="mt-2"
            >
              test-6
            </span>
          </div>
        </my-stack-stack>
        <my-stack-stack>
          <div
            class="flex flex-col items-center justify-items-center cursor-pointer"
          >
            <mat-card
              class="mat-mdc-card mdc-card h-[144px] w-[144px]"
            />
            <span
              class="mt-2"
            >
              test-7
            </span>
          </div>
        </my-stack-stack>
        <my-stack-stack>
          <div
            class="flex flex-col items-center justify-items-center cursor-pointer"
          >
            <mat-card
              class="mat-mdc-card mdc-card h-[144px] w-[144px]"
            />
            <span
              class="mt-2"
            >
              test-8
            </span>
          </div>
        </my-stack-stack>
        <my-stack-stack>
          <div
            class="flex flex-col items-center justify-items-center cursor-pointer"
          >
            <mat-card
              class="mat-mdc-card mdc-card h-[144px] w-[144px]"
            />
            <span
              class="mt-2"
            >
              test-9
            </span>
          </div>
        </my-stack-stack>
      </div>
    </div>
  </my-stack-stacks-overview>
</my-stack-stacks>
`;
